crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -76,6 +68,47 @@ pub(crate) struct ObjectKey {
     version: u64,
 }
 
+/// The object's owner type: Immutable, Shared, Parent, or Address.
+#[derive(Union, Clone)]
+pub enum ObjectOwner {
+    Immutable(Immutable),
+    Shared(Shared),
+    Parent(Parent),
+    Address(AddressOwner),
+}
+
+/// An immutable object is an object that can't be mutated, transferred, or deleted.
+/// Immutable objects have no owner, so anyone can use them.
+#[derive(SimpleObject, Clone)]
+pub struct Immutable {
+    #[graphql(name = "_")]
+    dummy: Option<bool>,
+}
+
+/// A shared object is an object that is shared using the 0x2::transfer::share_object function
+/// and is accessible to everyone.
+/// Unlike owned objects, once an object is shared, it stays mutable and can be accessed by anyone,
+/// unless it is made immutable. An example of immutable shared objects are all published packages
+/// and modules on Sui.
+#[derive(SimpleObject, Clone)]
+pub struct Shared {
+    initial_shared_version: u64,
+}
+
+/// The parent of this object
```

Comment:
nit: worth clarifying here that if the object's owner is a `Parent` in this sense, it's because this object is part of a dynamic field (it is the value of the dynamic field, or the intermediate `Field` object itself).

Reply:
I'll add this into a new PR. Thanks!